src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
Cool, thanks! I see the error. Your fix doesn't work, though, unless I'm missing something. The problem isn't so much in the array splicer, but in the cancel loop. Because cancel triggers a synchronous call to the doneCallbacks the splicer is called on the first element, removing it, and so the next loop tries to access and index that's not there anymore.
Reading the array from the end would work.